draft-ietf-quic-http.md

```diff
@@ -1286,15 +1286,15 @@ stream.  If the push stream has already ended, the server MAY still abruptly
 terminate the stream or MAY take no action.
 
 When a server sends CANCEL_PUSH, it is indicating that it will not be fulfilling
-a promise and has not created a push stream.  The client should not expect the
-corresponding promise to be fulfilled.
+a promise.  The client cannot expect the corresponding promise to be fulfilled.
+A server SHOULD send a CANCEL_PUSH even if it has opened the corresponding
+stream.
```

Comment:
Maybe not a direct result of this change but now I'm confused what happens if the server sends a CANCEL_PUSH that references a Push ID of a valid completed push stream. Do nothing seems like the right call.

Reply:
I think that is the right call.

Reply:
That matches what would occur in transport, so it does seem like the right call.

Reply:
Added text to this effect.  And that in fact could happen -- if we're suggesting a CANCEL_PUSH whenever the server abandons a stream (i.e. RESET_STREAM), it could be doing that to abort retransmissions even after writing the full payload.  The full payload might still have arrived if there's no loss.